src/main/scala/cromwell/binding/package.scala

```diff
@@ -30,4 +34,12 @@ package object binding {
     def serialize: Option[String] = attribute("serialize")
     def attribute(attr: String): Option[String] = runtimeAttributes.get(attr)
   }
+
+  /**
+   * Core data identifying a workflow including its unique ID, its namespace, and strongly typed inputs.
+   */
+  case class WorkflowDescriptor(namespace: WdlNamespace, actualInputs: WorkflowCoercedInputs) {
+    val id = UUID.randomUUID()
```

Comment:
Interesting, why did you choose to instantiate the UUID here?  I'm thinking mostly of the case where you want to recover an already running workflow after server restart, in that case, might you want to instantiate this with the UUID?